src/registry/radio/helpers.ts

```diff
@@ -9,3 +10,27 @@ export const checkIsManualOptions = (
 ): component is RadioComponentSchema & {values: Option[] | undefined} => {
   return component.openForms.dataSrc === 'manual';
 };
+
+// A type guard is needed because TS cannot figure out it's a discriminated union
+// when the discriminator is nested.
+// See https://github.com/microsoft/TypeScript/issues/18758
+export const checkIsReferentielijstenOptions = (
+  component: RadioComponentSchema
+): component is RadioComponentSchema & {
+  data: {values: Option[] | undefined};
+  openForms: {code: string; service: string};
+} => {
+  return component.openForms.dataSrc === 'referentielijsten';
```

Comment:
I think you can re-use the `ComponentWithReferentielijsten` type again here:

```ts
()component: RadioComponentSchema): component is RadioComponentSchema & ComponentWithReferentielijsten => {...}
```

Reply:
should I then still leave the `return component.openForms.dataSrc === 'referentielijsten';` part in there? It feels like both lines are stating the same thing then :thinking:

Reply:
yeah, that's how type guards work :) if the outcome of the runtime check is `true`, then the `component is FOO` thing applies (which narrows the type down to this particular discriminated union member). If it's `false`, all you know is that it's a `RadioComponentSchema` which can still be any of the other union members.

Reply:
ah like that, I was reading it the other way around, didn't know that the part after `:` was dependent on the return value, thanks for the explanation :+1: